reactioncommerce/reaction-component-library
# feat: fulfillmentOptionsCheckoutAction

Resolves #194 
Impact: **major**  
Type: **feature**





## Component

- [x] - Displays `availableFullfillmentOptions` in a `SelectableList`, or displays a `No methods` message
- [x] - Submits the selected fulfillment option
- [x] - Displays the selected fulfillment option
- [x] - Has cheapest selected by default

## Documentation
- Adds docs to component: https://deploy-preview-258--stoic-hodgkin-c0179e.netlify.com/#!/FulfillmentOptionsCheckoutAction
- Adds example in CheckoutAction doc https://deploy-preview-258--stoic-hodgkin-c0179e.netlify.com/#!/CheckoutAction

## Testing
1. Test the component itself: https://deploy-preview-258--stoic-hodgkin-c0179e.netlify.com/#!/FulfillmentOptionsCheckoutAction
2. Test it within CheckoutActions: https://deploy-preview-258--stoic-hodgkin-c0179e.netlify.com/#!/CheckoutActions

Test cases:
1. Make sure the cheapest is selected by default
2. Select and submit the cheapest by default
3. Select and submit a different one
4. Make sure the post-submit display message shows the correct open

## Screenshots

Post-completion success:
<img width="1055" alt="screen shot 2018-08-29 at 2 45 35 pm" src="https://user-images.githubusercontent.com/3673236/44817447-55f17180-ab9a-11e8-8270-095506b553fa.png">
